--- a/node_modules/jison-lex/regexp-lexer.js
+++ b/node_modules/jison-lex/regexp-lexer.js
@@ -405,12 +405,12 @@ RegExpLexer.prototype = {
 
     // return next match that has a token
     lex: function lex () {
-        var r = this.next();
-        if (r) {
-            return r;
-        } else {
-            return this.lex();
-        }
+        // https://github.com/zaach/jison-lex/pull/17
+        var r;
+        do {
+            r = this.next();
+        } while (!r);
+        return r;
     },
 
     // activates a new lexer condition state (pushes the new lexer condition state onto the condition stack)
